pallets/interest-accrual/src/lib.rs

```diff
@@ -241,47 +263,40 @@ pub mod pallet {
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
-			Rate::<T>::get(interest_rate_per_sec)
-				.ok_or(Error::<T>::NoSuchRate)
-				.and_then(|rate| {
-					Self::calculate_debt(normalized_debt, rate.accumulated_rate)
-						.ok_or(Error::<T>::DebtCalculationFailed)
-				})
-				.map_err(Into::into)
+			let rate = Self::get_rate(interest_rate_per_sec)?;
+			let debt = Self::calculate_debt(normalized_debt, rate.accumulated_rate)
+				.ok_or(Error::<T>::DebtCalculationFailed)?;
+			Ok(debt)
 		}
 
 		pub fn get_previous_debt(
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 			when: Moment,
 		) -> Result<T::Balance, DispatchError> {
-			if let Some(rate) = Rate::<T>::get(interest_rate_per_sec) {
-				let now = LastUpdated::<T>::get();
-				if when > now {
-					return Err(Error::<T>::NotInPast.into());
-				}
-				let delta = now - when;
-				let rate_adjustment = checked_pow(interest_rate_per_sec, delta as usize)
-					.ok_or(ArithmeticError::Overflow)?;
-				let past_rate = rate
-					.accumulated_rate
-					.checked_div(&rate_adjustment)
-					.ok_or(ArithmeticError::Underflow)?;
-				let debt = Self::calculate_debt(normalized_debt, past_rate)
-					.ok_or(Error::<T>::DebtCalculationFailed)?;
-				Ok(debt)
-			} else {
-				Err(Error::<T>::NoSuchRate.into())
+			let rate = Self::get_rate(interest_rate_per_sec)?;
+			let now = LastUpdated::<T>::get();
+			if when > now {
```

Comment:
I think `when` could be the same as `now` if the "previous debt" was computed in the same block (same `now` value)

Reply:
Yup, `when` can be ` now`, but the math still works out.

Reply:
Does the `checked_pow` has an early return path if `delta == 0`?

Reply:
> but the math still works out

Although the result should not change, right? 
I mean, as a loan user, do I want to get an error here if in the same block I call `borrow()` and then `repay()/reprice()/modify()`? Because my previous `last_update` will be used later in this function, but the value will have not changed. I do not know if I explain it well.

> Does the checked_pow has an early return path if delta == 0?
Yes (checked)